Tests/OnboardingTests/BindingPhoneNumberStateTests.swift

```diff
@@ -0,0 +1,352 @@
+    func testBindingPhoneNumber() async throws {
+        let stateMachine: StateMachine<BindingPhoneNumberState> = StateMachine(
+                initialState: .enterPhoneNumber(
+                        initialPhoneNumber: "1234567890",
+                        didSend: false,
+                        data: .init(seedPhrase: "", ethAddress: "", customShare: "", payload: "")
+                ),
+                provider: APIGatewayClientImplMock()
+        )
+
+        var nextState = try await stateMachine
+                .accept(event: .enterPhoneNumber(phoneNumber: "1234567890",
+                        channel: BindingPhoneNumberChannel.sms))
+        guard case .enterOTP = nextState else {
+            XCTFail("Expected .enterOTP, but was \(nextState)")
+            return
+        }
+    }
+
+    func testBindingPhoneNumber3() async throws {
+        let stateMachine: StateMachine<BindingPhoneNumberState> = StateMachine(
+                initialState: .enterPhoneNumber(initialPhoneNumber: "1234567890", didSend: true, data: .init(seedPhrase: "", ethAddress: "", customShare: "", payload: "")),
+                provider: APIGatewayClientImplMock()
+        )
+
+        var nextState = try await stateMachine
+                .accept(event: .enterPhoneNumber(phoneNumber: "1234567890", channel: .call))
+        guard case .enterOTP = nextState else {
+            XCTFail("Expected .enterOTP, but was \(nextState)")
+            return
+        }
+    }
+
+    func testBindingPhoneNumber4() async throws {
+        let stateMachine: StateMachine<BindingPhoneNumberState> = StateMachine(
+                initialState: .enterPhoneNumber(initialPhoneNumber: "1234567890", didSend: false, data: .init(seedPhrase: "", ethAddress: "", customShare: "", payload: "")),
+                provider: APIGatewayClientImplMock()
+        )
+        var phoneNumber = 0987654321
+        for i in 0...3 {
+            try await stateMachine
+                    .accept(event: .enterPhoneNumber(phoneNumber: String(phoneNumber + i), channel: .call))
+            try await stateMachine
+                    .accept(event: .back)
+        }
+        var nextState = try await stateMachine
+                .accept(event: .enterPhoneNumber(phoneNumber: "0987654325", channel: .call))
+        guard case .block = nextState else {
+            XCTFail("Expected .finish(.successful), but was \(nextState)")
+            return
+        }
+    }
+
+    func testBindingPhoneNumberEnterPhoneNumberError1() async throws {
+        let state: BindingPhoneNumberState = .enterPhoneNumber(initialPhoneNumber: "", didSend: false, data: .init(seedPhrase: "", ethAddress: "", customShare: "", payload: ""))
+
+        var nextState = try await state <- (
+                event: .enterPhoneNumber(phoneNumber: "1234532058", channel: .call),
+                provider: APIGatewayClientImplMock()
+        )
+        guard case .broken(code: -32058) = nextState else {
```

Comment:
does this test work? and can't see the final block state check. did I get it right?